popularity/lib/popularity.dart

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library package_popularity;
+
+import 'package:json_annotation/json_annotation.dart';
+
+part 'popularity.g.dart';
+
+@JsonSerializable()
+class PackagePopularity extends Object with _$PackagePopularitySerializerMixin {
+  @JsonKey(name: 'date_first')
+  final DateTime dateFirst;
+
+  @JsonKey(name: 'date_last')
+  final DateTime dateLast;
+
+  final List<PackageInfo> items;
```

Comment:
Since it's assumed we get exactly one `PackageInfo` per package name, which this `List` doesn't enforce, you could make it a `Map` instead with the keys being the package name.

Reply:
I hear ya. This file is effectively straight out of BigQuery...trying to minimize processing on our end. Will add a to-do

Reply:
...on second thought, I'll fix this.